lib/dry/cli/command.rb

```diff
@@ -13,37 +13,20 @@ class Command
       # @api private
       def self.inherited(base)
         super
+        base.class_eval do
+          @_mutex       = Mutex.new
+          @description  = nil
+          @examples     = []
+          @subcommands  = []
+          @arguments = base.superclass_variable_dup(:@arguments) || []
```

Comment:
@IvanShamatov IMO `superclass_variable_dup` is good as a private method to not be used directly outside of the class.

I would go for

```ruby
@arguments = base.superclass_arguments || []
```

```ruby
def self.superclass_arguments
  superclass_variable_dup(:@arguments)
end
```

Reply:
Ah, nice, will do that